cli_tools/gce_windows_upgrade/upgrader/upgrader.go

```diff
@@ -139,9 +148,59 @@ func (u *upgrader) init() error {
 	if err != nil {
 		return daisy.Errf("Failed to create GCE client: %v", err)
 	}
+
+	err = u.initLogger()
+	if err != nil {
+		return daisy.Errf("Failed to initialize logger: %v", err)
+	}
+
 	return nil
 }
 
+func (u *upgrader) initLogger() error {
+	loggingOptions := []option.ClientOption{option.WithCredentialsFile(u.Oauth)}
+	if !u.CloudLogsDisabled && u.ProjectPtr != nil {
+		cloudLoggingClient, err := logging.NewClient(u.ctx, *u.ProjectPtr, loggingOptions...)
+		if err != nil {
+			return err
+		}
+
+		if err := cloudLoggingClient.Ping(u.ctx); err != nil {
+			fmt.Println("Failed to initialize cloud logger: ", err)
+		} else {
+			cloudLogName := fmt.Sprintf("windows-upgrade-%s", u.id)
+			u.cloudLogger = cloudLoggingClient.Logger(cloudLogName)
+			go func() {
+				for {
+					time.Sleep(5 * time.Second)
+					u.cloudLogger.Flush()
+				}
+			}()
```

Comment:
How does the main execution ever finish with this infinite loop? Can you avoid this infinite loop?

Reply:
This side process won't block main process from exiting. Daisy logger did the same thing